pulsar-client/src/main/java/org/apache/pulsar/client/api/Consumer.java

```diff
@@ -294,4 +294,14 @@
      * @return consumer name.
      */
     String getConsumerName();
+
+    /**
+     * Stop requesting new messages from the broker until {@link #resume()} is called.
+     */
+    void pause();
```

Comment:
Barring the fact that pause/resume are clear in the context of listener, the only concern I have here is how `pause()` will interact with `receive()`. 
With this implementation, after calling `pause()`, a call to `receive()` might get blocked indefinitely. 

The question is should we allow this behavior or just make pause/resume to take effect on the listener. In either case, we should note that in this Javadoc.

Reply:
I have updated the Javadoc.